source/extensions/filters/common/rbac/matchers.cc

```diff
@@ -142,6 +145,34 @@ bool IPMatcher::matches(const Network::Connection& connection, const Envoy::Http
   case DownstreamLocal:
     ip = info.downstreamAddressProvider().localAddress();
     break;
+  case Upstream: {
+    if (!info.filterState().hasDataWithName(StreamInfo::KEY_DYNAMIC_PROXY_UPSTREAM_ADDR)) {
+      ENVOY_LOG_MISC(warn, "Did not find dynamic forward proxy metadata. Do you have dynamic "
+                           "forward proxy in the filter chain before the RBAC filter ?");
+      return false;
+    }
+
+    bool ipMatch = false;
+
+    const StreamInfo::AddressSetAccessor& address_set =
+        info.filterState().getDataReadOnly<StreamInfo::AddressSetAccessor>(
+            StreamInfo::KEY_DYNAMIC_PROXY_UPSTREAM_ADDR);
+
+    address_set.iterate([&, this](const Network::Address::InstanceConstSharedPtr& address) {
+      ipMatch = range_.isInRange(*address.get());
+      if (ipMatch) {
+        ENVOY_LOG_MISC(debug, "upstream_ip {} matched range: {}", address->asString(),
+                       range_.asString());
+        return false;
+      }
+
+      return true;
+    });
+
+    ENVOY_LOG_MISC(debug, "UpstreamIp matcher evaluated to: {}", ipMatch);
```

Comment:
would ENVOY_LOG_TO_LOGGER(Envoy::Logger::Registry::getLog(Envoy::Logger::Id::rbac) work over the MISC logger macro?